src/FSharp.Control.TaskSeq/TaskSeq.fsi

```diff
@@ -154,414 +175,872 @@ module TaskSeq =
+    /// Unboxes to the target type <see cref="'U" /> each item in the input task sequence.
     /// The target type must be a <see cref="struct" /> or a built-in value type.
     /// </summary>
+    ///
+    /// <param name="source">The input task sequence.</param>
+    /// <returns>The resulting task sequence.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input task sequence is null.</exception>
     /// <exception cref="InvalidCastException">Thrown when the function is unable to cast an item to the target type.</exception>
     val unbox<'U when 'U: struct> : source: taskSeq<obj> -> taskSeq<'U>
 
     /// <summary>
-    /// Casts each item in the untyped <paramref name="source" /> sequence asynchyronously. If your types are boxed struct types
+    /// Casts each item in the untyped input task sequence. If the input sequence contains value types
     /// it is recommended to use <see cref="TaskSeq.unbox" /> instead.
     /// </summary>
+    ///
+    /// <param name="source">The input task sequence.</param>
+    /// <returns>The resulting task sequence.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input task sequence is null.</exception>
     /// <exception cref="InvalidCastException">Thrown when the function is unable to cast an item to the target type.</exception>
-    val cast: source: taskSeq<obj> -> taskSeq<'T>
+    val cast: source: taskSeq<obj> -> taskSeq<'U>
 
-    /// Iterates over the taskSeq applying the action function to each item. This function is non-blocking
-    /// exhausts the sequence as soon as the task is evaluated.
+    /// <summary>
+    /// Iterates over the input task sequence, applying the <paramref name="action" /> function to each item.
+    /// This function is non-blocking, but will exhaust the full input sequence as soon as the task is evaluated.
+    /// </summary>
+    ///
+    /// <param name="action">A function to apply to each element of the task sequence.</param>
+    /// <param name="source">The input task sequence.</param>
+    /// <returns>A <see cref="unit" /> <see cref="task" />.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input sequence is null.</exception>
     val iter: action: ('T -> unit) -> source: taskSeq<'T> -> Task<unit>
 
-    /// Iterates over the taskSeq applying the action function to each item. This function is non-blocking,
-    /// exhausts the sequence as soon as the task is evaluated.
+    /// <summary>
+    /// Iterates over the input task sequence, applying the <paramref name="action" /> function to each item,
+    /// carrying the index as extra parameter for the <paramref name="action" /> function.
+    /// This function is non-blocking, but will exhaust the full input sequence as soon as the task is evaluated.
+    /// </summary>
+    ///
+    /// <param name="action">A function to apply to each element of the task sequence that can also access the current index.</param>
```

Comment:
This wording is not better yet, but 'can access' suggests that items intrinsically have an index and/or that there is some object wrapping going on, vs it being a temp maintained by the iteration process?
```suggestion
    /// <param name="action">A function to invoke per item of the task sequence, together with that items logical index in the sequence.</param>
```

Reply:
This is done for parity with `seq`. We should improve the docs there before doing it here.